src/View/SerializedView.php

```diff
@@ -84,8 +85,23 @@ public function render($view = null, $layout = null)
         }
 
         if ($serialize !== false) {
-            return $this->_serialize($serialize);
-        } elseif ($view !== false && $this->_getViewFileName($view)) {
+            $result = $this->_serialize($serialize);
+            if ($result === false) {
+                if (Configure::read('debug') && in_array('trace', $serialize)) {
```

Comment:
Yes, sure. But it is out of the hands of the developer in that case, and should at least try to avoid the exeption if possible (the other option of getting exceptions in your face and not knowing how to fix them is not really feasible). This piece is only relevant in debug mode, so non-debug mode is not affected by this change.